### PR TITLE
add url param `joinWithoutUserInteraction`

### PR DESCRIPTION
**Pull request by MaximeDucheneS**
**Use case**

In a webapp where users can move around multiple 'rooms' without ever refreshing their webpage, 
we do not want to click on 'Join the meeting' every time we change room,
to prevent unnecessary click and showing the same screen over and over.

**How**

By adding a new url param that allows to bypass the `ClickAnywhereToContinue` screen. This, with the following params, helps create the seamless experience that we need: 

```
micEnabled: false
webcamEnabled: false
joinScreenEnabled: false
```

**Testing**

Simply add the url param `joinWithoutUserInteraction` flag to your url.

If the join screen is disabled, you will immediately see the meeting.

### CODE REVIEW

**Comment by MaximeDucheneS** [comment]
@ChintanRajpara Hi! Arjun referred me to you for a review. Pleased to meet you!

**Comment by ChintanRajpara** [comment]
Hey @MaximeDucheneS, Thanks for contributing in our repository.

I had reviewed your changes, there are some changes that need to done for param name and setting the default value 

instead of `interactionOptional`, it should be renamed to  `userHasInteracted`. Default value of `userHasInteracted` will be `false`.

If `userHasInteracted` is `"true"` then set the state with the same name in App.js called `userHasInteracted`  to true else `null`

please add this changes. 

Thank you!

File: src/App.js [hunk 1]
```diff
@@ -294,6 +295,10 @@ const App = () => {
       playNotificationErr();
       setMeetingError({ message: configErr, code: 4001, isVisible: true });
     }
+    
+    if (paramKeys["interactionOptional"] === "true") {
+      setUserHasInteracted(true);
+    }
```
Comment (ChintanRajpara) [inline src/App.js:301]: no need of this if block

File: src/App.js [hunk 2]
```diff
@@ -102,6 +102,7 @@ const App = () => {
       reduceEdgeSpacing: "reduceEdgeSpacing",
       isRecorder: "isRecorder",
       leftScreenRejoinButtonEnabled: "leftScreenRejoinButtonEnabled",
+      interactionOptional: "interactionOptional",
```
Comment (ChintanRajpara) [inline src/App.js:105]: rename it to `userHasInteracted`.

**Comment by MaximeDucheneS** [comment]
@ChintanRajpara Thanks for the review!

**Pull request merged**